### ISSUE TITLE
Add possibility to use fields instead of column for unique constraint

### ISSUE THREAD

**Issue by alexander-schranz** [doctrine/orm#8345]
Currently when using a unique constraint like:

```xml
<unique-constraints>
    <unique-constraint columns="locale,product_key" />
</unique-constraints>
```

you are forced to define the `column` name for the fields e.g.:

```xml
<field name="productKey" column="product_key" type="string" nullable="true" unique="true"/>
```

if you don't do this and somebody configure another `naming_strategy` for your entity it will not longer work as the column  does then not exist.

It would be a great feature if you could use unique-constraints based on field name and doctrine orm will read the column name from the configured field e.g.:

```xml
<unique-constraints>
    <unique-constraint fields="locale,productKey" />
</unique-constraints>

<!-- no column is needed here now -->
<field name="productKey" type="string" nullable="true" unique="true"/>
```

What do you think about this?

**Comment by beberlei** [comment]
This is a great idea, there are some caviats. 

- What about associations? Is every join column added? That would make sense.
- Unique constraints need to be parsed after column/fields so that mapping info is available on CöassMetsdataInfo. This is not the case for all mapping drivers right now

**Comment by beberlei** [comment]
The logic of resolving Unique Constraint (and Index as well) fields to columns would need to be done in `SchemaTool`, because access to `NamingStrategy` is needed.

**Comment by Lustmored** [comment]
I took a shot at this and results are in #8523 - feedback is very welcome :)

**Comment by beberlei** [comment]
Fixed in #8629

**Comment by alexander-schranz** [comment]
@beberlei @Lustmored  This is really great to hear! Thank you for implementing this 🙏
